Pom/login

achuggi1 opened the pull request:
Created the LoginPage class and supporting files.

achuggi1 committed:
Checkpoint

Needed to move branches, so committed. Started with a rough outline of classes.

achuggi1 committed:
Skeleton LoginPage

Made a login page model complete with necessary IDs and names for the login elements. Still requires the login method to be implemented.

achuggi1 committed:
Implement login

Made a login implementation for the LoginPage.

Bryar-Frank commented on cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/Components/Component.java:
```diff
@@ -0,0 +1,13 @@
+
+package com.skillstorm.PageObjects.Components;
+
+public interface Component {
```
Should this have a more specific name than "Component"?

Bryar-Frank replied:
Mostly asking because Im not sure what this is supposed to do other than have a random "click button" method for now

achuggi1 replied:
I'd like a more specific name as well, not entirely sure what to name it though. I named it "Component" after React Components.

Bryar-Frank replied:
I see, can you explain what your idea is for the overarching concept of what this will do, that may give some more insight into naming it for its functionality... or is this just supposed to represent every possible component you can have in React?

achuggi1 replied:
It's supposed to be the highest level representation in the POM, and is implemented by things like Navbar and Page. I suppose a good way to put it would be that it represents every possible component you can have in React that has child components you can interact with.

msabas1 replied:
Seems like a generic React Components file for the different interactions with web elements, I think we could keep it as generic as Component.java for now

Bryar-Frank replied:
Since it is an interface though, this means that every object that represents a component MUST implement a method called "clickButton", but what if there is a component that doesn't need to click buttons, but wants to do something else... just seems too generic. Ill still approve it and all but it just doesn't make sense to me right now

Bryar-Frank replied:
resolving thought the question still remains lol

achuggi1 committed:
Add FindBy tags to LoginPage

Added the FindBy tags to WebElements that did not have an alternative loading option.

achuggi1 commented:
Closing request until Component work is done.
